### PR TITLE
Added Set Tags and Remove Tags to the exception list during sry run

### LINKED ISSUE #3985: Tags set using keywords don't appear in dryrun logs
Tags set using `set tags`, `pass execution`, `pass execution if` don't appear in dryrun log.

Run the following suite two times, using the command:
`robot -d normal tst_passExecTag.robot; robot -d dryrun --dryrun tst_passExecTag.robot`

```
*** settings ***
Documentation   Run with command robot --dryrun tst_passExecTag.robot
...
...    Run with command
...    ``robot -d normal tst_passExecTag.robot``
...    then with command
...    ``robot -d dryrun --dryrun tst_passExecTag.robot``

*** test cases ***
tst
    # This tag wont appear either
    set tags   tag set using set tags
    Pass execution  The following tag will appear in normal run mode but not in dryrun  tag set using pass execution
```

Then open the log.html files and compare their 'statistics by tag" section.

They differ, where I expect them to be the same.

This reduces the information level generated by dryrun.

### PR DESCRIPTION
**Pull request by vinayvennela**
Fixes #3985

### CODE REVIEW

File: src/robot/running/librarykeywordrunner.py
Comment (pekkaklarck) [inline src/robot/running/librarykeywordrunner.py:126]: Here seems to be a bug. This illustrates why all new features, like this one, need a test.

Extracting the list of keywords to a variable is a good idea. Perhaps it could just be named `keywords_to_execute`, though.

File: src/robot/running/librarykeywordrunner.py
Comment (pekkaklarck) [inline src/robot/running/librarykeywordrunner.py:221]: I don't think a space is needed here and there are some other style changes that I don't agree with. It would be better if this PR just added the feature and didn't touch style elsewhere.

**Comment by pekkaklarck** [comment]
Thanks for the PR! It would need tests, though. The suite to add the test is [here](https://github.com/robotframework/robotframework/blob/master/atest/robot/cli/dryrun/executed_builtin_keywords.robot), but you need to first read [atest/README.rst](https://github.com/robotframework/robotframework/blob/master/atest/README.rst), if you haven't already, to understand the test system.

There seems to also be a bug in the code that these tests ought to reveal.

Finally, it would be better if this PR just added the new functionality and didn't modify existing code.

**Comment by vinayvennela** [comment]
> Thanks for the PR! It would need tests, though. The suite to add the test is [here](https://github.com/robotframework/robotframework/blob/master/atest/robot/cli/dryrun/executed_builtin_keywords.robot), but you need to first read [atest/README.rst](https://github.com/robotframework/robotframework/blob/master/atest/README.rst), if you haven't already, to understand the test system.
> 
> There seems to also be a bug in the code that these tests ought to reveal.
> 
> Finally, it would be better if this PR just added the new functionality and didn't modify existing code.

Sure thing! Working on understanding the tests. 
I will update the PR soon with tests

**Comment by vinayvennela** [comment]
> > Thanks for the PR! It would need tests, though. The suite to add the test is [here](https://github.com/robotframework/robotframework/blob/master/atest/robot/cli/dryrun/executed_builtin_keywords.robot), but you need to first read [atest/README.rst](https://github.com/robotframework/robotframework/blob/master/atest/README.rst), if you haven't already, to understand the test system.
> > There seems to also be a bug in the code that these tests ought to reveal.
> > Finally, it would be better if this PR just added the new functionality and didn't modify existing code.
> 
> Sure thing! Working on understanding the tests.
> I will update the PR soon with tests

@pekkaklarck I have read the documentation and still unclear about how the atest/testdata and atest/robot are connected while running the tests. Can you give me some context around this.

I did add some tests to the atest/robot and they are running fine. What do I need to add in testdata folder how are they both connected?

**Comment by vinayvennela** [comment]
@pekkaklarck Updated the PR, please review

**Comment by vinayvennela** [comment]
Please review and let me know if more changes are needed

File: atest/robot/cli/dryrun/executed_builtin_keywords.robot
Comment (pekkaklarck) [inline atest/robot/cli/dryrun/executed_builtin_keywords.robot:22]: We have a separate `Check Test Tags` keyword that could be used to replace all the above like
```
Check Test Tags    ${TESTNAME}    Tag0    Tag1    Tag2    Tag3
```

File: atest/robot/cli/dryrun/executed_builtin_keywords.robot
Comment (pekkaklarck) [inline atest/robot/cli/dryrun/executed_builtin_keywords.robot:30]: `Check Test Tags` could be used here as well.

Extra empty lines at the end of the file should be removed. There should, however, be a new line at the end of the file.

File: atest/testdata/cli/dryrun/executed_builtin_keywords.robot
Comment (pekkaklarck) [inline atest/testdata/cli/dryrun/executed_builtin_keywords.robot:28]: Please add a newline here.

Yes, we are picky about whitespace usage. The reason is that otherwise there are unnecessary whitespace in diffs in the future.

File: src/robot/running/librarykeywordrunner.py
Comment (pekkaklarck) [inline src/robot/running/librarykeywordrunner.py:125]: Indentation looks strange. Names should be aligned like
```python
keywords_to_execute = ('BuiltIn.Import Library',
                       'BuiltIn.Set Library Search Order',
                       'BuiltIn.Set Tags',
                       'BuiltIn.Remove Tags')
```

Did I already mention we are picky about whitespace? :smile:

**Review by pekkaklarck — CHANGES_REQUESTED** [review]
Looks good but tests could be simplified a bit and whitespace usage fine-tuned. Hopefully you can do the changes!

File: atest/robot/cli/dryrun/executed_builtin_keywords.robot
Comment (vinayvennela) [reply atest/robot/cli/dryrun/executed_builtin_keywords.robot]: Thanks for the help! Updated

File: atest/robot/cli/dryrun/executed_builtin_keywords.robot
Comment (vinayvennela) [reply atest/robot/cli/dryrun/executed_builtin_keywords.robot]: Updated!

File: atest/testdata/cli/dryrun/executed_builtin_keywords.robot
Comment (vinayvennela) [reply atest/testdata/cli/dryrun/executed_builtin_keywords.robot]: Updated

File: atest/robot/cli/dryrun/executed_builtin_keywords.robot
Comment (pekkaklarck) [inline atest/robot/cli/dryrun/executed_builtin_keywords.robot:18]: This is actually unnecessary as `Check Test Tags` uses `Check Test Case` internally. I can remove these myself.

**Review by pekkaklarck — APPROVED** [review]

**Comment by pekkaklarck** [comment]
The PR looks good! I'll wait for CI to run tests and assuming there aren't problems I'll merge this. If you @vinayvennela are around, you can still remove the unnecessary `Check Test Case` lines from tests but I can do that after merging as well.

File: atest/robot/cli/dryrun/executed_builtin_keywords.robot
Comment (vinayvennela) [reply atest/robot/cli/dryrun/executed_builtin_keywords.robot:18]: I was referring to the existing tests and thought it was the convention. Thanks for the information 😃

**Pull request merged**
